lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -819,6 +819,100 @@ spec = do
         expectResponseCode @IO HTTP.status404 r
         expectErrorMessage (errMsg404NoWallet $ w ^. walletId) r
 
+    it "TRANS_ESTIMATE_08 - Fee estimation of edge cases" $ \ctx -> do
+        let walletBalance = 5000000
+        wSrc <- fixtureWalletWith @n ctx [walletBalance]
+        wDest <- emptyWallet ctx
+        addrs <- listAddresses @n ctx wDest
+        let destination = (addrs !! 1) ^. #id
+        let (feeMin, _) = ctx ^. #_feeEstimator $ PaymentDescription
+                { nInputs = 1
+                , nOutputs = 1
+                , nChanges = 1
+                }
+
+        let amt = walletBalance - feeMin :: Natural
+        let payload1 = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt},
+                        "unit": "lovelace"
+                    }
+                }]
+            }|]
+
+        r1 <- request @ApiFee ctx
+            (Link.getTransactionFee @'Shelley wSrc) Default payload1
+        verify r1
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#estimatedMin . #getQuantity)
+                (`shouldBe` feeMin)
+            ]
+
+        let payload2 = Json [json|{
+                "payments": [{
+                    "address": #{destination},
+                    "amount": {
+                        "quantity": #{amt+1},
+                        "unit": "lovelace"
+                    }
+                }]
+            }|]
+        r2 <- request @ApiFee ctx
+            (Link.getTransactionFee @'Shelley wSrc) Default payload2
+        verify r2
+            [ expectSuccess
+            , expectResponseCode HTTP.status202
+            , expectField (#estimatedMin . #getQuantity)
+                (`shouldBe` feeMin)
+            ]
```

Comment:
I do not understand what this second payload and second request are testing :thinking:  ?

Reply:
Actually, I am not sure I understand what the #3 and #4 are testing either actually?

Reply:
there are two thresholds that when triggered invoke different behavior. I write in detail about it in comment of this PR. basically they are set by The two thresholds are set by : 
```
         let (feeMin, _) = ctx ^. #_feeEstimator $ PaymentDescription                                                                                                                                          
                 { nInputs = 1                                                                                                                                                                                  
                 , nOutputs = 1                                                                                                                                                                                 
                 , nChanges = 1                                                                                                                                                                                 
                 }
```
and 
```
         let (feeMin', _) = ctx ^. #_feeEstimator $ PaymentDescription                                                                                                                                          
                 { nInputs = 1                                                                                                                                                                                  
                 , nOutputs = 1                                                                                                                                                                                 
                 , nChanges = 0                                                                                                                                                                                 
                 } 
```